### ISSUE TITLE
react/jsx-no-target-blank not working as expected on conditionals

### ISSUE THREAD

**Issue by akulsr0** [jsx-eslint/eslint-plugin-react#3462]
### Code

```jsx
const openInNewTab = true; // could be true or false

export default function App() {
  return (
    <a
      href="https://google.com"
      target={openInNewTab ? "_blank" : "self"}
      rel={openInNewTab ? "noreferrer" : undefined}
    >
      Open Google
    </a>
  );
}
```

### Currently what's happening

If I pass `target='_blank'` based on some condition as shown in the above snippet, and pass `rel='noreferrer'` on same condition. It is still showing me warning/error as per the config. 

### What should happen?

Ideally if the condition is same for both the props it should recognise it and should not give us warning/error.

Please let me know if I am missing something? 

### Reference:

Sample Codesandbox - https://codesandbox.io/s/mystifying-lake-l0420z?file=/src/App.js

**Comment by ljharb** [comment]
I agree, this would be a good enhancement to add. This seems highly similar to #2748, in case anyone wants to tackle it.
